arch/HPPA/HPPADisassembler.c

```diff
@@ -0,0 +1,3830 @@
+	case 0xc2:
+		MCInst_setOpcode(MI, HPPA_INS_MTCTL);
+		break;
+	case 0x45:
+		MCInst_setOpcode(MI, HPPA_INS_MFCTL);
+		if (get_insn_bit(insn, 17) == 1 &&
+		    MODE_IS_HPPA_20(MI->csh->mode)) {
+			push_str_modifier(HPPA_EXT_REF(MI), "w");
+		}
+		break;
+	}
+}
+
+static bool decode_sysop(const cs_struct *ud, MCInst *MI, uint32_t insn)
+{
+	uint32_t ext8 = get_insn_field(insn, 19, 26);
+	uint32_t ext5 = get_insn_field(insn, 11, 15);
+	uint32_t r1 = get_insn_field(insn, 6, 10);
+	uint32_t r2 = get_insn_field(insn, 11, 15);
+	uint32_t t = get_insn_field(insn, 27, 31);
+	uint32_t s = extract_3(insn);
+	if (MODE_IS_HPPA_20(MI->csh->mode)) {
+		switch (ext8) {
+		case 0xa5:
+			if (ext5 != 0) {
+				return false;
+			}
+			CREATE_GR_REG(MI, t);
+			return true;
+		case 0xc6:
+			CREATE_GR_REG(MI, r2);
+			return true;
+		}
+	}
+
+	switch (ext8) {
+	case 0x00:
+		MCOperand_CreateImm0(MI, t);
+		MCOperand_CreateImm0(MI, get_insn_field(insn, 6, 18));
+		return true;
+	case 0x20:
+		if (ext5 != 0x00 && ext5 != 0x10) {
+			return false;
+		}
+		// fallthrough
+	case 0x60:
+	case 0x65:
+		return true;
+	case 0x6b:
+	case 0x73:
+		MCOperand_CreateImm0(MI, get_insn_field(insn, 9, 15));
+		CREATE_GR_REG(MI, t);
+		return true;
+	case 0xc3:
+		CREATE_GR_REG(MI, r2);
+		return true;
+	case 0x85:
+		CREATE_SR_REG(MI, s);
+		CREATE_GR_REG(MI, r1);
+		CREATE_GR_REG(MI, t);
+		return true;
+	case 0xc1:
+		CREATE_GR_REG(MI, r2);
+		CREATE_SR_REG(MI, s);
+		return true;
+	case 0x25:
+		if (ext5 != 0) {
+			return false;
+		}
+		CREATE_SR_REG(MI, s);
+		CREATE_GR_REG(MI, t);
+		return true;
+	case 0xc2:
+		CREATE_GR_REG(MI, r2);
+		CREATE_CR_REG(MI, r1);
+		return true;
+	case 0x45:
+		if (ext5 != 0) {
+			return false;
+		}
+		if (get_insn_bit(insn, 17) == 1 && MODE_IS_HPPA_20(ud->mode) &&
+		    r1 != 11) {
+			return false;
+		}
+		CREATE_CR_REG(MI, r1);
+		CREATE_GR_REG(MI, t);
+		return true;
+	default:
+		return false;
+	}
+}
+
+static void fill_memmgmt_insn_name(MCInst *MI, uint32_t insn)
+{
+	uint32_t ext = get_insn_field(insn, 19, 25);
+	if (MODE_IS_HPPA_20(MI->csh->mode)) {
+		switch (ext) {
+		case 0x20:
+			MCInst_setOpcode(MI, HPPA_INS_IITLBT);
+			return;
+		case 0x18:
+			MCInst_setOpcode(MI, HPPA_INS_PITLB);
+			push_str_modifier(HPPA_EXT_REF(MI), "l");
+			return;
+		case 0x60:
+			MCInst_setOpcode(MI, HPPA_INS_IDTLBT);
+			return;
+		case 0x58:
+			MCInst_setOpcode(MI, HPPA_INS_PDTLB);
+			push_str_modifier(HPPA_EXT_REF(MI), "l");
+			return;
+		case 0x4f:
+			MCInst_setOpcode(MI, HPPA_INS_FIC);
+			return;
+		case 0x46:
+			if (get_insn_bit(insn, 18) == 0) {
+				MCInst_setOpcode(MI, HPPA_INS_PROBE);
+			} else {
+				MCInst_setOpcode(MI, HPPA_INS_PROBEI);
+			};
+			push_str_modifier(HPPA_EXT_REF(MI), "r");
+			return;
+		case 0x47:
+			if (get_insn_bit(insn, 18) == 0) {
+				MCInst_setOpcode(MI, HPPA_INS_PROBE);
+			} else {
+				MCInst_setOpcode(MI, HPPA_INS_PROBEI);
+			};
+			push_str_modifier(HPPA_EXT_REF(MI), "w");
+			return;
+		}
```

Comment:
This switch still misses the default case. Please add it, because some compilers will complain.